javascript/lib.c

```diff
@@ -17,20 +18,20 @@ napi_value FindAPI(napi_env env, napi_callback_info info) {
     napi_get_cb_info(env, info, &argc, args, NULL, NULL);
 
     // Extract the C string from the JavaScript string for haystack and needle
-    size_t str_size;
-    size_t str_len;
+    size_t haystack_l;
+    size_t needle_l;
 
     // For haystack
-    napi_get_value_string_utf8(env, args[0], NULL, 0, &str_size);
-    char *haystack = malloc(str_size + 1);
-    napi_get_value_string_utf8(env, args[0], haystack, str_size + 1, &str_len);
-    struct strzl_haystack_t strzl_haystack = {haystack, str_len};
+    napi_get_value_string_utf8(env, args[0], NULL, 0, &haystack_l);
+    char *haystack = malloc(haystack_l + 1);
+    napi_get_value_string_utf8(env, args[0], haystack, haystack_l + 1, &needle_l);
```

Comment:
Pass haystack_l here again. It will be overwritten by the same value, which is fine, but we will not confuse the purpose of variables.